lib/SILOptimizer/Mandatory/SILGenCleanup.cpp

```diff
@@ -104,39 +108,219 @@ struct SILGenCleanup : SILModuleTransform {
   void run() override;
 
   bool completeOSSALifetimes(SILFunction *function);
+  template <typename Range>
+  bool completeLifetimesInRange(Range const &range,
+                                OSSALifetimeCompletion &completion,
+                                BasicBlockSet &completed);
 };
 
+// Iterate over `iterator` until finding a block in `include` and not in
+// `exclude`.
+SILBasicBlock *findFirstBlock(SILFunction::iterator &iterator,
+                              BasicBlockSet const &include,
```

Comment:
You could also directly pass `BasicBlockWorklist`s for `include` and `exclude`. Then you don't need to add `BasicBlockWorklist::getVisited()`

Reply:
I prefer this approach.  I think this method should exist anyway.

Reply:
> I think this method should exist anyway.

I don't agree because it's an implementation detail of BasicBlockWorklist that `isVisited` is implemented as a set. Exposing this detail makes BasicBlockWorklist a leaking abstraction.

Reply:
I don't think this constitutes any meaningful abstraction leakage (a worklist is going to use a set), and I think we it should be possible to get read-only access to the set that is produced via the worklist to pass off to other code that operates on sets.

Reply:
I replaced uses of the sets with function_refs wrapping calls to isVisited.